python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -830,6 +836,55 @@ def addDimensionEntry(self, dimension, dataId=None, entry=None, **kwds):
             self.setDimensionRegion(dataId)
         return dataId
 
+    @disableWhenLimited
+    @transactional
+    def addDimensionEntryList(self, dimension, dataIdList, entry=None, **kwds):
```

Comment:
Is there a downside to `addDimensionEntry` calling this itself so you don't get the duplicated code?  Also it seems odd to me that the list version has significantly more in it than the one-at-a-time version.

Reply:
I agree.  This also duplicates some of the `setDimensionRegion` code, so perhaps that needs a `List` version as well to handle the `skypix`/region handling?

Reply:
The handling of setting dimension regions is a tiny bit different when calling the two different functions. In the future there will be a ticket to vectorize many of the other registry calls, in which case I agree that the single entry could differ to the vectorized version.